src/libstore/filetransfer.cc

```diff
@@ -216,6 +216,19 @@ struct curlFileTransfer : public FileTransfer
                         encoding = trim(line.substr(i + 1));
                     else if (name == "accept-ranges" && toLower(trim(line.substr(i + 1))) == "bytes")
                         acceptRanges = true;
+                    else if (name == "x-amz-meta-nix-is-immutable" || name == "x-nix-is-immutable") {
```

Comment:
I also wonder if it might be more accurate to be named or `x-nix-redirect-is-stable` or something to make it more obvious that this is describing the fact that the redirect target isn't expected to change, not that its contents aren't expected to change (a slight difference).

But maybe this is probably bikeshedding, so feel free to disregard.

Reply:
As a practical issue, using a header in the redirect doesn't work with S3 website buckets, since they don't allow custom headers on redirects.

Reply:
Given https://www.rfc-editor.org/rfc/rfc6648, I'd recommend not having an `X-` prefix on the header.